plugins/modules/rudder_node_settings.py

```diff
@@ -280,6 +280,44 @@ def _send_request(
                     status_code, resp.content
                 ),
             )
+            
+    def check_NodeSettingValue(self, node_id, property_name, property_value):
+        """Function to check if a setting or property as already set on node via the API
+        
+        Args:
+            values (str): Values to check in API
+            property_name (str): Name of your node property to check
+            property_value (str): Value of your node property to check
+        
+        Returns:
+            bool: Return if yes (true) or no (false) a value as already set (default: False)
+        """
+        url = "/api/latest/nodes/{node_id}\?include=default".format(node_id=self.node_id)
```

Comment:
I feel like the node_id must not be a class attribute, since the playbook offers the possibility to use queries instead of an id. You want to have all your functions independent of the attributes node_id, otherwise you will have to implement some unwanted logic to separate those cases or instanciate a new object for each node